staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -189,6 +194,46 @@ func GetClientIP(req *http.Request) net.IP {
 	return net.ParseIP(req.RemoteAddr)
 }
 
+// Prepares the X-Forwarded-For header for another forwarding hop by appending the previous senders
+// IP address to the X-Forwarded-For chain.
+func PrepareForwardedFor(req *http.Request) {
+	remoteIP := GetRemoteIP(req)
+
+	hdr := req.Header
+	// First check the X-Forwarded-For header for requests via proxy.
+	hdrForwardedFor := hdr.Get("X-Forwarded-For")
+	if hdrForwardedFor != "" {
+		// Append the remote IP if necessary.
+		parts := strings.Split(hdrForwardedFor, ",")
+		for i := len(parts) - 1; i >= 0; i-- {
+			// Find the last non-empty part.
+			if parts[i] == "" {
+				continue
+			}
+			if ip := net.ParseIP(strings.TrimSpace(parts[i])); ip.Equal(remoteIP) {
+				// Remote IP already appended; nothing to do.
+				return
+			}
+			parts = append(parts[:i+1], remoteIP.String())
+			hdr.Set("X-Forwarded-For", strings.Join(parts, ","))
+			return
```

Comment:
Returning at the end of the main for loop seems a but clumsy. Can we find a nicer way to strip trailing commas?